wifistats/index.js

```diff
@@ -0,0 +1,129 @@
+ *
+ * @callback callback delivering results
+ * @param {Error} error if any
+ * @param {object} result stats
+ */
+function initTools(callback) {
+    async.parallel([
+            function (cb) {
+                getStats(
+                    function (err, str) {
+                        cb(null, {err: err, tool: airport, result: str});
+                    }, airport);
+            },
+            function (cb) {
+                getStats(function (err, str) {
+                    cb(null, {err: err, tool: procwireless, result: str});
+                }, procwireless);
+            },
+            function (cb) {
+                getStats(function (err, str) {
+                    cb(null, {err: err, tool: netsh, result: str});
+                }, netsh);
+            }
+        ],
+        function (err, results) {
+            const res = results.find((r) => (r.err == null));
+            if (res) {
+                callback(null, res.tool, res.result);
+            } else {
+                callback(new Error('No known wifi stats tool found'));
+            }
+        }
+    );
+}
+
+/**
+ * Queries for wifi stats using a specific tool or if not specified, uses the
+ * global one which we discover to be working.
+ * @param tool - The tool to use querying for wifi stats.
+ * @callback callback delivering results
+ * @param {Error} error if any
+ * @param {object} result stats
+ */
+function getStats(callback, tool) {
+    if (!tool) {
+        if (toolInstance) {
+            tool = toolInstance;
+        } else {
+            callback(new Error('No known wifi stats tool found'));
+        }
+    }
+
+    exec(tool.cmdLine, function (err, str) {
+        if (err) {
+            callback(err, null);
+            return;
+        }
+
+        tool.parseOutput(str, callback);
+    });
+}
+
+/**
+ * Queries for wifi stats.
+ * @callback callback delivering results
+ * @param {Error} error if any
+ * @param {object} result stats
+ * @returns {Promise}
+ */
+function getWiFiStats() {
+    return new Promise((resolve, reject) => {
+        if (!toolInstance) {
+            initTools(
+                (err, t, result) => {
+                    if (err) {
+                        reject(err);
+                    }
+                    toolInstance = t;
+                    resolve(result);
+                });
+            return;
+        }
+
+        getStats((error, result) => {
+            if (error) {
+                reject(error);
+            } else {
+                resolve(JSON.stringify(result));
+            }
+        });
+    });
+}
+
+/**
+ * Setup getWiFiStats to be available in JitsiMeetElectron object.
+ *
+ * @param iframe - the iframe to use attaching the getWiFiStats function.
+ */
+function setupWiFiStats(iframe) {
+    iframe.addEventListener('load', () => {
+        const ctx = iframe.contentWindow;
+        if(typeof ctx.JitsiMeetJS === "undefined")
+            ctx.JitsiMeetJS = {};
```

Comment:
In what cases do you expect this error to be encountered? Is there a danger of the global getting overridden?

Reply:
Well, as not sure when this will be called we do not want to override the namespace if it exists. I borrowed the code from jitsi-meet.

Reply:
I'm more concerned about the global you declared getting overridden by the lib which I assume creates the global JitsiMeetJS as well. That currently does not happen, right? I'm assuming it doesn't happen.

Reply:
Yep, it is working. 
Ok, let's discuss this offline tomorrow, cause I'm really confused about this. As I had exposed the function directly in the window, but everybody says that we want to move away from that and expose everything under common namespace and advice me to use this namespace the way we do in jitsi-meet. So your concern is also valid and for jitsi-meet. Maybe I had messed up something and didn't understand.